src/renderer/PresetsBar.tsx

```diff
@@ -84,48 +90,51 @@ const PresetListItem = ({
 };
 
 const PresetsBar = () => {
-  const { globalError } = useAquaContext();
+  const { globalError, performHealthCheck, setGlobalError } = useAquaContext();
 
   const [presetName, setPresetName] = useState<string>('');
   const [selectedPresetName, setSelectedPresetName] = useState<
     string | undefined
   >(undefined);
 
-  // TODO: Fetch default presets and custom presets from storage
-  const [presetNames, setPresetNames] = useState<string[]>([
-    'Airpods Pro to IEF Neutral AutoEQ',
-    'Airpods to Harman AE OE 2018 AutoEQ',
-    '7-1 Surround Sound Speaker Setup',
-    'Bass Boost',
-    'Bass and Treble Boost',
-    'Classic',
-    'Dance',
-    'Rock',
-    'Vocal Boost',
-    'Microphone Compensation',
-    'MORE BASS',
-    'Surround effect',
-    'Indie playlist',
-    'Pop',
-    'Hip-hop',
-    'R&B',
-    'Equalize for shrill noise',
-    'Jazz',
-  ]);
-
-  const handleCreatePreset = (prev: string[]) => {
-    if (prev.indexOf(presetName) !== -1) {
-      // TODO: overwrite preset values for the existing preset
-    } else {
-      // TODO: add code for creating a preset
+  const [presetNames, setPresetNames] = useState<string[]>([]);
+
+  // Fetch default presets and custom presets from storage
+  useEffect(() => {
+    const fetchPresetNames = async () => {
+      const result = await getPresetListFromFiles();
```

Comment:
Need to handle exception case

Reply:
right now just populating with empty list. We could do a popup or something? not sure how the error handling should be handled here

Reply:
@xenown mentioned in the gc already, but you can set the global error by calling `setGlobalError(e);` - this will open up the `PrereqMissingModal` (which, in retrospect, we should really rename it to something more general like `ErrorModal`) that prompts users to either retry (aka refresh the page) or exit the app.